// @aws-sdk/client-lex-runtime-v2/dist-es/commands/StartConversationCommand.js
import { getEventStreamPlugin } from "@aws-sdk/middleware-eventstream";
import { getEndpointPlugin } from "@smithy/middleware-endpoint";
import { getSerdePlugin } from "@smithy/middleware-serde";
import { Command as $Command } from "@smithy/smithy-client";
import { SMITHY_CONTEXT_KEY, } from "@smithy/types";
import { StartConversationRequestFilterSensitiveLog, StartConversationResponseFilterSensitiveLog, } from "../models/models_0";
import { de_StartConversationCommand, se_StartConversationCommand } from "../protocols/Aws_restJson1";
export { $Command };
export class StartConversationCommand extends $Command {
    static getEndpointParameterInstructions() {
        return {
            UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
            Endpoint: { type: "builtInParams", name: "endpoint" },
            Region: { type: "builtInParams", name: "region" },
            UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
        };
    }
    constructor(input) {
        super();
        this.input = input;
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointPlugin(configuration, StartConversationCommand.getEndpointParameterInstructions()));
        this.middlewareStack.use(getEventStreamPlugin(configuration));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "LexRuntimeV2Client";
        const commandName = "StartConversationCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: StartConversationRequestFilterSensitiveLog,
            outputFilterSensitiveLog: StartConversationResponseFilterSensitiveLog,
            [SMITHY_CONTEXT_KEY]: {
                service: "AWSDeepSenseRunTimeServiceApi2_0",
                operation: "StartConversation",
            },
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return se_StartConversationCommand(input, context);
    }
    deserialize(output, context) {
        return de_StartConversationCommand(output, context);
    }
}
