// @aws-sdk/client-lex-runtime-v2/dist-cjs/LexRuntimeV2.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LexRuntimeV2 = void 0;
const smithy_client_1 = require("@smithy/smithy-client");
const DeleteSessionCommand_1 = require("./commands/DeleteSessionCommand");
const GetSessionCommand_1 = require("./commands/GetSessionCommand");
const PutSessionCommand_1 = require("./commands/PutSessionCommand");
const RecognizeTextCommand_1 = require("./commands/RecognizeTextCommand");
const RecognizeUtteranceCommand_1 = require("./commands/RecognizeUtteranceCommand");
const StartConversationCommand_1 = require("./commands/StartConversationCommand");
const LexRuntimeV2Client_1 = require("./LexRuntimeV2Client");
const commands = {
    DeleteSessionCommand: DeleteSessionCommand_1.DeleteSessionCommand,
    GetSessionCommand: GetSessionCommand_1.GetSessionCommand,
    PutSessionCommand: PutSessionCommand_1.PutSessionCommand,
    RecognizeTextCommand: RecognizeTextCommand_1.RecognizeTextCommand,
    RecognizeUtteranceCommand: RecognizeUtteranceCommand_1.RecognizeUtteranceCommand,
    StartConversationCommand: StartConversationCommand_1.StartConversationCommand,
};
class LexRuntimeV2 extends LexRuntimeV2Client_1.LexRuntimeV2Client {
}
exports.LexRuntimeV2 = LexRuntimeV2;
(0, smithy_client_1.createAggregatedClient)(commands, LexRuntimeV2);
